### ISSUE TITLE
Update the examples to include the db name in the sync url instead of the collection name

### ISSUE THREAD

**Issue by rmisio** [pubkey/rxdb#1007]
Currently the examples include the collection name in the sync url, for example:

```
  collections
    .filter(col => col.sync)
    .map(col => col.name)
    .map(colName => {
      return db[colName].sync({
        remote: syncUrl + colName + '/'
```

The issue with that is then different local databases end up writing to a single database named as the collection name.

I'm guessing the specifics of the url must have changed since the examples were written.

The correct way I believe to do it now is to include the db name in the url and the collection name is derived from the fact that you're calling sync on the collection instance:

```
  collections
    .filter(col => col.sync)
    .map(col => col.name)
    .map(colName => {
      return db[colName].sync({
        remote: `${syncUrl}/${dbName}/`,
```

**Comment by pubkey** [comment]
This does not look correct for me.
If you have a local collection `user`, then you want to sync it with the remote couchdb-instance `user`.
If you have collection `comments` then you want to sync it with the remote `comments` instance.

With your change, all collections would sync with the same remote couchdb-instance which will then mix up the user and the comments.

**Comment by rmisio** [comment]
Well, when I use this url  `remote: syncUrl + colName + '/'`, then in the fauxton UI for the sync db, in the `database name` column, I end up getting the collection name.

`profile` is the name of my collection here:

![image](https://user-images.githubusercontent.com/794517/55654789-8b54e200-57af-11e9-8d35-06433491541d.png)

Furthermore, when two seperate databases write to the `profile` collection, the documents both end up in that `profile` link above.

Am I misunderstanding the Fauxton UI, because it seems like it's grouping documents from different DB's into one single database.

These `profile` documents were created from 2 different DB's.

![image](https://user-images.githubusercontent.com/794517/55654891-d3740480-57af-11e9-8a4e-7b7021405368.png)

**Comment by rmisio** [comment]
@pubkey Is there contact information for you anywhere? I wanted some more detail on what the different Patreon levels would get me as far as support.

**Comment by pubkey** [comment]
I'm closing this, because I think this is a problem related to the understanding of fauxton.

A RxCollection is equal to a pouchdb-database. This means many RxCollection are many pouchdb-databases. You should replicate a single RxCollection to a single pouchdb-database and therefore set the collection-name in the sync-url.
Syncing multiple collections with the same sync-url does not make sense to me.

**Comment by rmisio** [comment]
> A RxCollection is equal to a pouchdb-database. 

hmmm... I thought a collection is analogous to a table in a standard DB in which case it's absolutely a very common thing for a database to have multiple tables?

**Comment by pubkey** [comment]
Yes, in RxDB a collection is similar to a table in a SQL-Database. Equal to how it works in mongodb.

In Pouchdb and Couchdb, there is no concept of table. There are only databases. This is a more granular approach which is sometime error-prone when data is aggregated over multiple databases. This is why I have chosen to use collections in rxdb. Logically they work like tables, but technically they are instances of pouchdb.

**Comment by rmisio** [comment]
The main problem here is this:

- create db1
- create a collection on db1 named 'happy'
- add a document to the collection
- set up syncing with the documented url format of `remote: syncUrl + colName + '/'`.
- create db2
- create a collection on db2 named 'happy'
- add a document to the collection
- set up syncing with the documented url format of `remote: syncUrl + colName + '/'`.
- fetch an item from the happy collection of db1.
- *both added documents are in the collection*

something is being messed up where the collections with the same name in two different DB's get their documents co-mingled.

As far as I can tell, this is something that broke in the newer version of RxDB because in the older version this doesn't happen.

**Comment by pubkey** [comment]
If you do not want to mix the collection from db1 and db2, you should not set the same sync-target.
You can prefix the sync-url with the db-name if you want.

But I do not think this should be changed in the docs. There only the collection-name is set, so it matches with the sync-urls which are generated by the [server-plugin](https://rxdb.info/tutorials/server.html)

**Comment by pubkey** [comment]
Ah wait. They are not. Let me check.

**Comment by rmisio** [comment]
> If you do not want to mix the collection from db1 and db2, you should not set the same sync-target.
You can prefix the sync-url with the db-name if you want.

ok, but just so you're aware this behavior changed in the last major rxdb version. In a previous version it somehow managed to split things between DBs.

I assume since the sync call is made on the db instance, it knows which db we're trying to sync.

`db[colData.name].sync({`

**Comment by pubkey** [comment]
@rmisio are you sure that this worked before version 8?
In the last major, not that much has changed in the sync-behavior. Setting the same sync-location from multiple collections should also have mixed up the documents

**Comment by rmisio** [comment]
@pubkey I'm sure. I wrote a proof of concept of our web version a few months back. Part of our reqs are multi-tenancy where the web app supports multiple users that have a local database that's syncable to a central db-server.

There was no issue switching between users.

And... as a sanity check when I logged this bug back in April I went back and checked and I made sure the documented sync url was being used.
